t/plugin/response-rewrite2.t

```diff
@@ -517,3 +517,181 @@ passed
+                    "uris": ["/hello"]
+                }]]
+                )
+
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 21: hit
+--- request
+GET /hello
+--- response_headers
+Cache-Control: no-cache, max-age=0, must-revalidate
+
+
+
+=== TEST 22: set headers
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "plugins": {
+                        "response-rewrite": {
+                            "headers": {
+                                "add": [
+                                    "Cache-Control: no-cache"
+                                ],
+                                "set": {
+                                    "Cache-Control": "max-age=0, must-revalidate"
+                                }
+                            }
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uris": ["/hello"]
+                }]]
+                )
+
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 23: hit
+--- request
+GET /hello
+--- response_headers
+Cache-Control: max-age=0, must-revalidate
+
+
+
+=== TEST 24: remove headers
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "plugins": {
+                        "response-rewrite": {
+                            "headers": {
+                                "add": [
+                                    "Set-Cookie: <cookie-name>=<cookie-value>; Max-Age=<number>"
+                                ],
+                                "set": {
+                                    "Cache-Control": "max-age=0, must-revalidate"
+                                },
+                                "remove": [
+                                    "Set-Cookie",
+                                    "Cache-Control"
+                                ]
+                            }
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uris": ["/hello"]
+                }]]
+                )
+
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 25: hit
+--- request
+GET /hello
+--- response_headers
+Cache-Control:
+Set-Cookie:
```

Comment:
Shouldn't these two response headers appear? instead of just the value being null.

Reply:
We usually use `name:` in `response_headers` to check the header `name` doesn't exist.

Reply:
Is this the rule for test-nginx? Got it.